main/lsp/requests/code_action.cc

```diff
@@ -260,6 +346,76 @@ unique_ptr<ResponseMessage> CodeActionTask::runRequest(LSPTypecheckerDelegate &t
             action->kind = CodeActionKind::RefactorRewrite;
             action->edit = move(workspaceEdit);
             result.emplace_back(move(action));
+        } else {
+            ExtractVariableWalk extractVariableWalk(loc);
+            auto desugaredTree = typechecker.getDesugared(loc.file());
+            core::Context ctx(gs, core::Symbols::root(), loc.file());
+            ast::TreeWalk::apply(ctx, extractVariableWalk, desugaredTree);
+
+            if (extractVariableWalk.getFoundExactMatch()) {
+                auto tdi =
+                    make_unique<VersionedTextDocumentIdentifier>(move(params->textDocument->uri), JSONNullObject());
+                vector<unique_ptr<TextEdit>> edits;
+
+                core::Loc whereToInsert;
```

Comment:
You can simplify this a little if whereToInsert is a LocOffsets and construct the `Loc` only once.